tracer.go

```diff
@@ -175,12 +175,16 @@ func (t *tracer) startSpanWithOptions(
 
 	var samplerTags []tag
 	var ctx SpanContext
-	if !hasParent {
+	debugRequest := (hasParent && parent.traceID == 0 && parent.debugID != "")
```

Comment:
it's a bit of a hack, but I didn't know how to do it better. The problem is that previously if headers do not have trace context, the extract() method would return ErrNoSpanContextFound. But if jaeger-debug-id is set, it now returns an otherwise empty SpanContext with non-empty debugID. So really the SpanContext is used as a sentinel & container of debugID, and this is what I am verifying here. Maybe I should move it to a method on the SpanContext?